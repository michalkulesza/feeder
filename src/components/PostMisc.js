import React from "react";

const PostMisc = ({ active, handleNewPost }) => {
  return (
    <div className={active ? "post-misc active" : "post-misc"}>
      <span></span>

      <div className="right">
        <div className="button" onClick={handleNewPost}>
          Send it
        </div>
      </div>
    </div>
  );
};

export default PostMisc;
