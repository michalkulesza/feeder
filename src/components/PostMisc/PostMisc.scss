@import "../../main.scss";

.post-misc {
	height: 0px;
	transform: scaleY(0);
	width: 100%;
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin-top: 0px;
	transition: all 0.2s ease-in-out;
	&.active {
		margin-top: 12px;
		height: 33px;
		transform: scaleY(1);
	}

	&.right {
		width: 80px;
		height: 40px;

		&.button {
			background-color: pink !important;
		}
	}
}
