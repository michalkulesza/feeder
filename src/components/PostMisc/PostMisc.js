import React from "react";
import "./PostMisc.scss";

const PostMisc = ({ active, handleNewPost }) => {
  return (
    <div className={active ? "post-misc active" : "post-misc"}>
      <span></span>

      <div className="right">
        <div className="button green" onClick={handleNewPost}>
          Send it
        </div>
      </div>
    </div>
  );
};

export default PostMisc;
