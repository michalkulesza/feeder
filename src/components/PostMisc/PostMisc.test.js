import React from "react";
import { shallow } from "enzyme";

import PostMisc from "./PostMisc";

describe("<PostMisc/>", () => {
	const handleNewPost = jest.fn();

	it("should match the snapshot without props", () => {
		const wrapper = shallow(<PostMisc />);

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should match the snapshot with props", () => {
		const active = false;
		const wrapper = shallow(<PostMisc active={active} handleNewPost={handleNewPost} />);

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should render component with active class", () => {
		const active = true;
		const wrapper = shallow(<PostMisc active={active} handleNewPost={handleNewPost} />);
		const activeClass = wrapper.find(".active");

		expect(activeClass).toBeTruthy();
	});

	it("should call the click handler", () => {
		const active = true;
		const wrapper = shallow(<PostMisc active={active} handleNewPost={handleNewPost} />);
		const button = wrapper.find(".button");

		button.simulate("click");
		expect(handleNewPost).toBeCalled();
	});
});
