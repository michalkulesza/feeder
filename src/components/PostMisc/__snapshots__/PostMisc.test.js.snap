// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostMisc/> should match the snapshot with props 1`] = `"<div class=\\"post-misc\\"><span></span><div class=\\"right\\"><div class=\\"button green\\">Send it</div></div></div>"`;

exports[`<PostMisc/> should match the snapshot without props 1`] = `"<div class=\\"post-misc\\"><span></span><div class=\\"right\\"><div class=\\"button green\\">Send it</div></div></div>"`;
